src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_rsa.cpp

```diff
@@ -267,3 +267,76 @@ static int32_t ExecuteCFDataTransform(
 
     return ret;
 }
+
+static int32_t RsaPrimitive(SecKeyRef key,
+                            uint8_t* pbData,
+                            int32_t cbData,
+                            CFDataRef* pDataOut,
+                            CFErrorRef* pErrorOut,
+                            SecKeyAlgorithm algorithm,
+                            CFDataRef func(SecKeyRef, SecKeyAlgorithm, CFDataRef, CFErrorRef*))
+{
+    if (pDataOut != nullptr)
+        *pDataOut = nullptr;
+    if (pErrorOut != nullptr)
+        *pErrorOut = nullptr;
+
+    if (key == nullptr || pbData == nullptr || cbData < 0 || pDataOut == nullptr || pErrorOut == nullptr)
+    {
+        return kErrorBadInput;
+    }
+
+    assert(func != nullptr);
```

Comment:
Just curious, why is this an assert rather than verified in the above return kErrorBadInput check?

Reply:
The above checks are for values that came across the P/Invoke boundary, so it's an API constraint enforcement.  `func` has to be passed in via a non-C-static function in this library, so it's an internal consistency check instead of an API constraint